### PR TITLE
[FLINK-11483][tests] Improve StreamOperatorSnapshotRestoreTest with JUnit's Parameterized

### PR DESCRIPTION
**Pull request by klion26**
## What is the purpose of the change

Improve StreamOperatorSnapshotRestoreTest with JUnit's Parameterized

## Brief change log

In current we will test StreamOperatorSnapshotRestore with three backends: `file`, `rocksdb full` and `rocksdb incremental`, each in a separate class.
In the current patch, we improve this by using JUnit's Parameterized.

## Verifying this change

This change is already covered by existing tests, such as `StreamOperatorSnapshotRestoreTest`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**no**)
  - The serializers: (**no**)
  - The runtime per-record code paths (performance sensitive): (**no**)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (**no**)
  - The S3 file system connector: (**no**)

## Documentation

  - Does this pull request introduce a new feature? (**no**)
  - If yes, how is the feature documented? (**not applicable**)

### CODE REVIEW

File: flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/StreamOperatorSnapshotRestoreTest.java [hunk 1]
```diff
@@ -80,6 +94,18 @@
 
 	protected static TemporaryFolder temporaryFolder;
 
+	@Parameterized.Parameter
+	public StateBackendEnum stateBackendEnum;
+
+	enum StateBackendEnum {
```
[the remark this answers is not in the document]
Comment (klion26) [reply flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/StreamOperatorSnapshotRestoreTest.java]: Here we use `Enum` to iterate the threes statebackens, we have to return the StateBackend Instance in `parameter()` if we don't have the `Enum`. Please correct me if I'm wrong.

**Comment by klion26** [comment]
@tzulitai thank you for your quick review. I think you may have pinged the wrong person in the last comment.

**Comment by tzulitai** [comment]
Thanks, fixed the ping.

File: flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/StreamOperatorSnapshotRestoreTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tzulitai) [reply flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/StreamOperatorSnapshotRestoreTest.java]: I meant that we don't necessarily have to have the word `Enum` in the name, seems redundant.
More of a nitpick only :)

[the remark this answers is not in the document]
Comment (klion26) [reply flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/StreamOperatorSnapshotRestoreTest.java]: Thank you for your patient explanation. If we change `enum StateBackendEnum` to `enum StateBackend`, then we have to replace `StateBackend` with `org.apache.flink.runtime.state.StateBackend` in line 162 and line 235, so I chose the `enum StateBackendEnum` style.

**Comment by StefanRRichter** [comment]
@tzulitai changing the ping seemingly did not work, but now I saw it anyways :-) I have no objections against the refactoring, looks good to me. 👍

**Pull request merged**